elastic/apm-server
# model: move HTTP to APMEvent, unify fields

## Motivation/summary

Move `model.HTTP` out of `model.Transaction`, `model.Span`, and `model.Error`, and into `model.APMEvent`. We now set the same top-level `http.*` fields for both transactions and spans. Some fields are copied for spans in addition, for backwards compatibility.

This change means that some extended (non-ECS) span HTTP fields are now recorded under both `http.*` and `span.http.*`. The latter will be removed in 8.0, in https://github.com/elastic/apm-server/issues/5995.

## Checklist

~- [ ] Documentation has been updated~

## How to test these changes

1. Send some spans with non-ECS HTTP fields (e.g. `testadata/intake-v2/spans.ndjson`)
2. Search for a span with `span.http.response.encoded_body_size`, check that it has the field duplicated as `http.response.encoded_body_size`.

## Related issues

#4120 
#5995
